android-core/src/main/java/com/mparticle/internal/MessageHandler.kt

```diff
@@ -0,0 +1,331 @@
+package com.mparticle.internal
+
+import android.content.Context
+import android.os.Looper
+import android.os.Message
+import com.mparticle.MParticle
+import com.mparticle.internal.Constants.MessageKey
+import com.mparticle.internal.MPUtility.addNumbers
+import com.mparticle.internal.MPUtility.isEmpty
+import com.mparticle.internal.MParticleApiClientImpl.MPNoConfigException
+import com.mparticle.internal.MessageManager.IncrementUserAttributeMessage
+import com.mparticle.internal.MessageManager.ReportingMpidMessage
+import com.mparticle.internal.database.services.MParticleDBManager
+import com.mparticle.internal.database.services.MParticleDBManager.AttributionChange
+import com.mparticle.internal.database.services.MParticleDBManager.UserAttributeRemoval
+import com.mparticle.internal.database.services.MParticleDBManager.UserAttributeResponse
+import com.mparticle.internal.database.tables.SessionTable
+import com.mparticle.internal.messages.BaseMPMessage
+import com.mparticle.internal.messages.MPAliasMessage
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.AbstractMap
+import java.util.UUID
+
+/* package-private */
+internal open class MessageHandler : BaseHandler {
+    private val mContext: Context
+
+    var mMParticleDBManager: MParticleDBManager
+
+    private val mMessageManagerCallbacks: MessageManagerCallbacks
+    var mDataplanId: String? = null
+    var mDataplanVersion: Int? = null
+
+    /**
+     * for unit testing only
+     */
+    constructor(
+        messageManager: MessageManagerCallbacks,
+        context: Context,
+        dbManager: MParticleDBManager,
+        dataplanId: String?,
+        dataplanVersion: Int?
+    ) {
+        mMessageManagerCallbacks = messageManager
+        mContext = context
+        mMParticleDBManager = dbManager
+        mDataplanId = dataplanId
+        if (dataplanVersion != null) {
+            mDataplanVersion = dataplanVersion
+        }
+    }
+
+    constructor(
+        looper: Looper,
+        messageManager: MessageManagerCallbacks,
+        context: Context,
+        dbManager: MParticleDBManager,
+        dataplanId: String?,
+        dataplanVersion: Int?
+    ) : super(looper) {
+        mMessageManagerCallbacks = messageManager
+        mContext = context
+        mMParticleDBManager = dbManager
+        if (dataplanId != null) {
+            mDataplanId = dataplanId
+        }
+        if (dataplanVersion != null) {
+            mDataplanVersion = dataplanVersion
+        }
```

Comment:
```suggestion
        mDataplanId = dataplanId
        mDataplanVersion = dataplanVersion
```

Same here, these are both nullable so no null check needed.